Pds4File/pds4_rules/cassini_iss.py

```diff
@@ -292,10 +292,18 @@
 # OPUS_ID
 ####################################################################################################################################
 
+#opus_id = translator.TranslatorByRegex([
+#    (r'.*/COISS_[12]xxx.*/([NW][0-9]{10})_[0-9]+.*', 0, r'co-iss-#LOWER#\1'),
+#])
+
 opus_id = translator.TranslatorByRegex([
-    (r'.*/COISS_[12]xxx.*/([NW][0-9]{10})_[0-9]+.*', 0, r'co-iss-#LOWER#\1'),
+     (r'.*/cassini_iss/cassini_iss\w*/data_raw/\d{3}xxxxxxx/\d{5}xxxxx/(\d{10})(n|w).[a-z]{3}', 0, r'co-iss-\2\1')
 ])
```

Comment:
I think one OPUS ID is associated with both a data_raw and a browse_raw product (e.g.  co-iss-w1308947228 is associated with data_raw/130xxxxxxx/13089xxxxx/1308947228w.img and browse_raw/130xxxxxxx/13089xxxxx/1308947228w-full.png), and so could use either?

Reply:
I think there is some confusion about this. The `opus_id` translator is supposed to handle ANY product file and return the associated OPUS ID. Give it a browse image, a .qub file, a .lbl file, a calibrated file, whatever, and it will return the appropriate OPUS ID. So it has to handle ALL of the possible associated products. The original PdsFile handles this as:

```
opus_id = translator.TranslatorByRegex([
    # There are up to two OPUS IDs associated with each VIMS file, one for the VIS channel and one for the IR channel.
    # This translator returns the OPUS ID without the suffix "_IR" or "_VIS" used by OPUS. That must be handled separately
    (r'.*/COVIMS_0xxx.*/(v[0-9]{10})_[0-9]+(|_[0-9]{3})\..*', 0, r'co-vims-\1\2'),
])
```

Note the `.*` at the beginning. The handles the `volumes`, `previews`, and `metadata` directories. This is easy in the PDS3 case since the directory structures are parallel among the different product types.

Reply:
I did get myself confused! Thanks for the clarification